independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -255,14 +256,21 @@ static boolean isObjectToString(MethodInfo method) {
             if (Modifier.isPrivate(method.flags())
                     && !Annotations.contains(methodAnnotations, DotNames.OBSERVES)
                     && !Annotations.contains(methodAnnotations, DotNames.OBSERVES_ASYNC)) {
```

Comment:
I think we should also add a check for `@Produces`...

Reply:
Done